### PR TITLE
Make jenkins green... again

### PR DESCRIPTION
**Pull request by rheacangeo**
Fixing various issues that are disrupting our tests, including:
* replace cached_property in MetricTerms with a decorator that works, as the former was caching at the class level, not the instance level, impacting validation when the tests are run together
* adjusting some thresholds that need changing for running standalone, for grid computation on c48
* add a --skip_grid_tests flag to test_regressions that enables turning off testing grid generation. This was added so on PRs we can not run these tests (since the code is unlikely to change, and if impacted by other changes, other code is to likely fail as well) because they increase the time to compute (especially on c54 ranks) significantly. We will not turn off testing the grid in the regressions-cron plan so that someone is alerted if a code change breaks grid generation
* fixed the state initialization to work on gpu, doing numpy operations on the memoryview does not work
* run_on_daint script takes a namelist and an experiment name rather than a whole dataset, since we don't need the serialized data for this script anymore. run_standalone passes the namelist and experiment name to it.

### CODE REVIEW

**Comment by rheacangeo** [comment]
launch jenkins

File: fv3core/fv3core/grid/generation.py [hunk 1]
```diff
@@ -44,20 +44,10 @@
...
-    @functools.wraps(func)
-    def wrapped(*args, **kwargs):
-        nonlocal cached
-        if cached is None:
-            cached = func(*args, **kwargs)
-        return cached
-
-    return wrapped
-
+# TODO: when every environment in python3.8, replace
```
Comment (twicki) [inline fv3core/fv3core/grid/generation.py:47]: we could even have:
```
def cached_property(f):
    @property
    @functools.lru_cache()
    def wrapper(self, *args, **kwargs):
        return f(self, *args, **kwargs)
    return wrapper
```
here and then the only change would be to get rid of it once we've moved up. Not super necessary

File: fv3core/.jenkins/actions/run_standalone.sh [hunk 2]
```diff
@@ -134,18 +135,11 @@ if [ "${SAVE_TIMINGS}" == "true" ] && [ "${SAVE_ARTIFACTS}" == "true" ] ; then
...
-    find .gt_cache* -type d -name \*_pyext_BUILD -prune -exec \rm -rf {} \;
-    find .gt_cache* -type d -name __pycache__ -prune -exec \rm -rf {} \;
-    echo "Copying GT4Py cache directories to ${CACHE_DIR}"
-    mkdir -p ${CACHE_DIR}
-    cp ${ROOT_DIR}/GT4PY_VERSION.txt ${CACHE_DIR}
-    rm -rf ${CACHE_DIR}/.gt_cache*
-    cp -rp .gt_cache* ${CACHE_DIR}
+# copying the cache is in a separate action (generache_cache.sh),
+# otherwise delete it
+if [ "${SAVE_CACHE}" != "true" ] ; then
```
[the remark this answers is not in the document]
Comment (rheacangeo) [reply fv3core/.jenkins/actions/run_standalone.sh:140]: I was trying to leave open the flexibility to call other code, such as the regression tests, and then at the end the jenkins plan would call the generate_caches script. I think run_standalone should be sufficient for the cache, but have you seen evidence more things are compiled when we run the regression tests?

File: fv3core/examples/standalone/runfile/dynamics.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def parse_args() -> Namespace:
         "data_dir",
         type=str,
         action="store",
-        help="directory containing data to run with",
+        help="namelist configuration for model run",
```
[the remark this answers is not in the document]
Comment (rheacangeo) [reply fv3core/examples/standalone/runfile/dynamics.py:42]: good catch, danger of manually undoing a change!

File: fv3core/examples/standalone/runfile/dynamics.py [hunk 4]
```diff
@@ -154,6 +154,37 @@ def gather_hit_counts(
     return results
 
 
+def get_experiment_info(data_directory: str) -> Tuple[str, bool]:
+    config_yml = yaml.safe_load(
+        open(
+            data_directory + "/input.yml",
```
[the remark this answers is not in the document]
Comment (rheacangeo) [reply fv3core/examples/standalone/runfile/dynamics.py:160]: they should both be with the data now. We can eventually refer to one, but the Namelist class got a lot more hard coded... it's a bit of a hassle to check it for the test_case which might not exist. It's easier to read this for the moment. 'experiment' can indeed be derived from the folder or the yml

File: fv3core/examples/standalone/runfile/dynamics.py [hunk 5]
```diff
@@ -154,6 +154,37 @@ def gather_hit_counts(
...
+def read_serialized_initial_state(rank):
+    # set up of helper structures
+    serializer = serialbox.Serializer(
+        serialbox.OpenModeKind.Read,
+        args.data_dir,
+        "Generator_rank" + str(rank),
+    )
+    # create a state from serialized data
+    savepoint_in = serializer.get_savepoint("FVDynamics-In")[0]
+    driver_object = fv3core.testing.TranslateFVDynamics([grid])
```
[the remark this answers is not in the document]
Comment (rheacangeo) [reply fv3core/examples/standalone/runfile/dynamics.py:182]: lol how did this work at all?

File: fv3core/fv3core/grid/generation.py (hunk 1 above)
Comment (rheacangeo) [reply fv3core/fv3core/grid/generation.py:47]: I got really sick of trying out decorators and them not working right... I can do try this if you want

**Review by twicki — APPROVED** [review]
lgtm

**Pull request merged**